### PR TITLE
[QoL] Add setting to control volume of UI sound efffects

### PR DESCRIPTION
**Pull request by tlelievre26**
## What are the changes?
Adds an option to the settings menu that toggles the sound effects from the UI.

## Why am I doing these changes?
Relevant issue: https://github.com/pagefaultgames/pokerogue/issues/3164

## What did change?

Adds a new variable to the settings menu that controls the volume of sounds originating from menus. This mostly means the "select" sound, along with the sounds for opening/closing some menus.

### Screenshots/Videos

image

## How to test the changes?

Manually tested by enabling the setting and moving around the main menu, starter select menu, and playing through a few battles to ensure all UI sounds were disabled

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Adri1** [comment]
Could you add it to all other locales please?

**Comment by tlelievre26** [comment]
> Could you add it to all other locales please?

Done

File: src/locales/fr/settings.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ export const settings: SimpleTranslationEntries = {
   "masterVolume": "Vol. principal",
   "bgmVolume": "Vol. musique",
   "seVolume": "Vol. effets",
+  "uiSoundEffects": "UI Sound Effects",
```
Comment (Adri1) [inline src/locales/fr/settings.ts:66]:
```suggestion
  "uiSoundEffects": "Son des menus",
```

File: src/locales/ko/settings.ts [hunk 2]
```diff
@@ -63,6 +63,7 @@ export const settings: SimpleTranslationEntries = {
   "masterVolume": "마스터 볼륨",
   "bgmVolume": "BGM 볼륨",
   "seVolume": "SE 볼륨",
+  "uiSoundEffects": "UI Sound Effects",
```
Comment (sodaMelon) [inline src/locales/ko/settings.ts:66]:
```suggestion
  "uiSoundEffects": "UI 효과음",
```

**Review by sodaMelon — COMMENTED** [review]
added korean localization

**Review by 40chyan — COMMENTED** [review]
zh

File: src/locales/de/settings.ts [hunk 3]
```diff
@@ -63,6 +63,7 @@ export const settings: SimpleTranslationEntries = {
   "masterVolume": "Gesamtlautstärke",
   "bgmVolume": "Hintergrundmusik",
   "seVolume": "Spezialeffekte",
+  "uiSoundEffects": "UI Sound Effects",
```
Comment (CodeTappert) [inline src/locales/de/settings.ts:66]:
```suggestion
  "uiSoundEffects": "UI-Soundeffekte",
```

File: src/locales/pt_BR/settings.ts [hunk 4]
```diff
@@ -63,6 +63,7 @@ export const settings: SimpleTranslationEntries = {
   "masterVolume": "Volume Mestre",
   "bgmVolume": "Volume de BGM",
   "seVolume": "Volume de SE",
+  "uiSoundEffects": "UI Sound Effects",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/settings.ts:66]:
```suggestion
  "uiSoundEffects": "Efeitos Sonoros do Menu",
```

**Comment by EnochG1** [comment]
@tlelievre26 
From now Japanese is added in beta.
Could you merge latest beta and add Japanese locale files? 🙏

**Comment by tlelievre26** [comment]
I think I'm still waiting for the Spanish + Japanese translation correct?

File: src/locales/es/settings.ts [hunk 5]
```diff
@@ -64,6 +64,7 @@ export const settings: SimpleTranslationEntries = {
   "masterVolume": "Master Volume",
   "bgmVolume": "BGM Volume",
   "seVolume": "SE Volume",
+  "uiSoundEffects": "UI Sound Effects",
   "musicPreference": "Music Preference",
   "mixed": "Mixed",
   "gamepadPleasePlug": "Please Plug in a Gamepad or Press a Button",
```
Comment (AsdarDevelops) [inline src/locales/es/settings.ts:70]:
```suggestion
  "masterVolume": "Volumen general",
  "bgmVolume": "Volumen de la música",
  "seVolume": "Volumen de los efectos",
  "uiSoundEffects": "Volumen de la interfaz",
  "musicPreference": "Preferencia de música",
  "mixed": "Mixto",
  "gamepadPleasePlug": "Conecta un Gamepad o pulsa un botón",
```

**Review by AsdarDevelops — COMMENTED** [review]
Added also some other translations that were missing in Spanish

**Comment by torranx** [comment]
To be consistent with other sound controls, I'd prefer this to be an actual volume control just like the others.

File: src/locales/it/settings.ts [hunk 6]
```diff
@@ -64,6 +64,7 @@ export const settings: SimpleTranslationEntries = {
   "masterVolume": "Master Volume",
   "bgmVolume": "BGM Volume",
   "seVolume": "SE Volume",
+  "uiSoundEffects": "UI Sound Effects",
```
Comment (NicusPulcis) [inline src/locales/it/settings.ts:67]:
```suggestion
  "uiSoundEffects": "Effetti sonori UI",
```

**Review by NicusPulcis — APPROVED** [review]
italian localization is done

File: src/ui-inputs.ts [hunk 7]
```diff
@@ -181,7 +181,9 @@ export class UiInputs {
       break;
     case Mode.MENU:
       this.scene.ui.revertMode();
-      this.scene.playSound("select");
+      if (this.scene.uiSoundEffects) {
```
Comment (frutescens) [inline src/ui-inputs.ts:184]: Please move this check to inside the playSound() method in battle-scene.ts

File: src/ui/menu-ui-handler.ts [hunk 8]
```diff
@@ -321,7 +321,9 @@ export default class MenuUiHandler extends MessageUiHandler {
 
     this.getUi().hideTooltip();
 
-    this.scene.playSound("menu_open");
+    if (this.scene.uiSoundEffects) {
+      this.scene.playSound("menu_open");
```
Comment (frutescens) [inline src/ui/menu-ui-handler.ts:325]: See above.

File: src/ui/ui.ts [hunk 9]
```diff
@@ -395,11 +395,15 @@ export default class UI extends Phaser.GameObjects.Container {
   }
 
   playSelect(): void {
-    (this.scene as BattleScene).playSound("select");
+    if ((this.scene as BattleScene).uiSoundEffects) {
```
Comment (frutescens) [inline src/ui/ui.ts:398]: See above comment.

File: src/ui/ui.ts [hunk 10]
```diff
@@ -395,11 +395,15 @@ export default class UI extends Phaser.GameObjects.Container {
...
+      (this.scene as BattleScene).playSound("select");
+    }
   }
 
   playError(): void {
-    (this.scene as BattleScene).playSound("error");
+    if ((this.scene as BattleScene).uiSoundEffects) {
```
Comment (frutescens) [inline src/ui/ui.ts:404]: See above comment.

**Review by frutescens — CHANGES_REQUESTED** [review]
Hi. Thank you for writing this PR. Please let me know if you have any questions about battle-scene.ts ' playSound() function.

**Comment by tlelievre26** [comment]
> Hi. Thank you for writing this PR. Please let me know if you have any questions about battle-scene.ts ' playSound() function.

Thanks for pointing this out, I refactored this PR to work with the changes introduced in #3527, which includes changing this setting from a toggle to a volume control.

Only things still worth considering:
Should there be any other audio files considered as "ui" sounds other than "select", "error", and "menu_open"
Should the name of this setting be changed to "UI Volume" to be consistent with the other settings? I don't want to make the translators redo all their work

**Comment by CodeTappert** [comment]
You have conflicts that need fixing

**Review by CodeTappert — APPROVED** [review]

File: src/battle-scene.ts [hunk 11]
```diff
@@ -1799,11 +1800,14 @@ export default class BattleScene extends SceneBase {
           config["volume"] *= 0.5;
         }
         break;
-      case "se":
       case "ui":
+        config["volume"] = this.masterVolume * this.uiVolume;
```
Comment (frutescens) [inline src/battle-scene.ts:1804]: can you add a comment specifying the sounds under the UI label? have you decided to make any decisions re: the categorization from before?

**Review by frutescens — CHANGES_REQUESTED** [review]

**Comment by torranx** [comment]
Please update PR title and description

**Review by Madmadness65 — APPROVED** [review]

**Review by torranx — APPROVED** [review]

**Review by frutescens — APPROVED** [review]

**Review by SangaraSorama — APPROVED** [review]
My review is only about the locales/en file

**Pull request merged**